gems/aws-sigv4/spec/aws-sigv4/signer_spec.rb

```diff
@@ -345,25 +339,51 @@ def initialize(credentials)
       end
 
       describe '#sign_request' do
+        context 'request object' do
```

Comment:
I think we'll want this in presigned_url too, both the test AND the logic. I believe how the specs are today, they are kind of duplicative, across the methods. Could possibly consider a shared example - but we should ensure request is handled in that case too.

Reply:
Added to presigned_url tests.  Shared examples might be a little tricky - there are a few differences in the methods that make it hard to really abstract it.